x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -323,3 +336,24 @@ export function updateDrawState(drawState: DrawState | null) {
     });
   };
 }
+
+export function removeFeaturesFromIndexQueue(featureIds: string[]) {
+  return {
+    type: REMOVE_FEATURES_FROM_INDEX_QUEUE,
+    featureIds,
+  };
+}
+
+export function setVectorLayerIndexName(indexName: string) {
+  return {
+    type: SET_VECTOR_LAYER_INDEX_NAME,
+    indexName,
+  };
+}
+
+export function addNewFeatureToIndex(indexName: string, geometry: unknown, path: string) {
```

Comment:
make `addNewFeatureToLayer(layerId, geometry: Geometry);`

add async-methods `IVectorLayer.addFeature` and `ESSearchSource.addFeature`, and move implementations there.  (they can use the utils of course).

Will help having to pass all these params around (`indexName`, `path`) and keep that locally inside the `ESSearchSource` implementation.